opslevel/resource_opslevel_service.go

```diff
@@ -302,26 +302,14 @@ func resourceServiceUpdate(d *schema.ResourceData, client *opslevel.Client) erro
 	id := d.Id()
 
 	input := opslevel.ServiceUpdateInput{
-		Id: opslevel.ID(id),
-	}
-
-	if d.HasChange("name") {
```

Comment:
This is what the input looks like in opslevel-go:

```
type ServiceUpdateInput struct {
	Id          ID               `json:"id,omitempty"`
	Alias       string           `json:"alias,omitempty"`
	Name        string           `json:"name,omitempty"`
	Product     string           `json:"product,omitempty"`
	Description string           `json:"description,omitempty"`
	Language    string           `json:"language,omitempty"`
	Framework   string           `json:"framework,omitempty"`
	Tier        string           `json:"tierAlias,omitempty"`
	Owner       *IdentifierInput `json:"ownerInput,omitempty"`
	Lifecycle   string           `json:"lifecycleAlias,omitempty"`
}
```

The d.Get() function from terraform will return an empty string for a key that is in the schema if it is not set in the configuration. Therefore, `""` will be set as the field which is a non-pointer, and since `omitempty` is enabled for the field the ...

https://pkg.go.dev/github.com/hashicorp/terraform/helper/schema#ResourceData.Get